<?xml version="1.0"?>
<launch>

	<node pkg="tf2_ros" type="static_transform_publisher" name="base_to_camera" args="0 0 0.4 0 0 0 base_link camera_link"/>
	<node pkg="tf2_ros" type="static_transform_publisher" name="world_to_image_coordinate_system" args="0 0 0 0.5 -0.5 0.5 0.5 worldcoordinates cameracoordinates"/>





	<!-- Nodes to be launched -->



	<!--<include file="$( find bounding_box_filter)/launch/bounding_box_filter.launch"/>-->
	<node name="smoreo" pkg="smoreo" type="smoreo_system.py" output="screen"/>



					<!-- Camera_parameters-->
	<param name="image_width"    								value="1440"/>
	<param name="image_height"    							value="1080"/>
	<param name="smoreo/f"    							value="720.47"/>
	<param name="hfov"    									value="90"/>
	<param name="smoreo/cx"		    						value="720.0"/>
	<param name="smoreo/cy"    								value="560.01"/>
	<param name="smoreo/pixel_size"    							value="0.00345"/>

						<!-- Published topics -->
	<param name="/smoreo/predicted_landmarks" 		value="/perception/smoreo/detected_cones_lm"/>
						<!-- Subscribed topics -->
	<param name="/smoreo/bounding_boxes" 				value="/darknet_ros/bounding_boxes"/>
	<param name = "/smoreo/camera_info"					value = "/front_camera_rgb/camera_info" />

	<param name="/smoreo/use_filtered" 					value="False"/>
	<param name="/smoreo/hardcode_params" 				value="True"/>
	<param name="/smoreo/cone_height" 					value="0.38"/>
	<param name="/smoreo/cut_off_y" 					value="900.0"/>
	<param name="/smoreo/use_cone_base" 				value= "True"/>
	<param name="/smoreo/camera_height_from_ground" 	value="0.816"/>
</launch>
